### PR TITLE
[form recognizer] add receipt locale tests

### LINKED ISSUE #13461: [formrecognizer] add tests for receipt locale

### PR DESCRIPTION
**Pull request by iscai-msft**
fixes #13461

### CODE REVIEW

**Review by kristapratico — DISMISSED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_from_url_async.py
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_from_url_async.py:415]: sorry last minute, but should we add a test here that test locale with api version 2.0?

Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt_from_url_async.py:415]: oof good point. i copied these over from business cards, where it was not necessary

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py
Comment (maririos) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:429]: I'm trying to understand this. So we are doing a call and we want to see if the service returned `en-US`? or does we set a default in the SDK>

Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:429]: basically, I'm trying to confirm if the user doesn't pass in `locale`, the default locale that is being used is in the call to the service is `en-US`. While this checks more the swagger / our generation of it, I wanted to include it since we are saying in our docstring that the default is `en-US`

Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:429]: actually @maririos i realized based off of your comment that my tests aren't actually working, since the default isn't set in the swagger, it's set on the service side. so going to redo my checking of these tests

Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:429]: but we do want to set it in the SDK as default, right? just verifying

Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:429]: no, we should leave that up to the service. Ignore everything I said about swagger, since it's not actually set as the default in the swagger. The service is saying (in their docs at least): that if you don't send us `locale` in your request in a query parameter, we're going to use our default of 'en-US'

Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:429]: My test actually isn't correct, because my callback where I'm checkign the query parameters in the call to the service actually is never called, so please ignore this test 😆

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py
Comment (kristapratico) [inline sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py:441]: sorry didn't notice this before, let's not assert on error messages since they can change. Asserting on error code should be fine.

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py
Comment (maririos) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py]: This goes against what Java is doing. Before merging this PR let's make sure that our SDKs are aligned 
FYI @samvaity

Comment (iscai-msft) [reply sdk/formrecognizer/azure-ai-formrecognizer/tests/test_receipt.py]: @maririos @samvaity merging bc paul has p much decided this for us 😄

**Pull request merged**
